api/server/router/system/system_routes.go

```diff
@@ -57,6 +77,15 @@ func (s *systemRouter) swarmStatus() string {
 	return string(swarm.LocalNodeStateInactive)
 }
 
+func (s *systemRouter) handleCapabilities(ctx context.Context, capabilitiesVersion int) (capabilities.VersionedCapabilities, error) {
+	log.G(ctx).WithField("version", capabilitiesVersion).Debug("capabilities requested")
+	caps, err := s.backend.SystemCapabilities(ctx, capabilitiesVersion)
+	if err != nil {
+		return nil, err
+	}
+	return caps, nil
+}
+
```

Comment:
Probably mostly a nit, but I was trying to see what this wrapper added as extra beyond the debug log;

- it looks like _currently_ `s.backend.SystemCapabilities` never returns an error
- ☝️ was this added to handle future errors for not-supported versions?
- if we don't have a good purpose yet; we could even consider dropping the error-return; the backend interfaces are mostly considered "internal" interfaces to decouple the API, so making signature changes there should be fine.

For the logs; possibly the existing logs capture some of that already (perhaps the "fetching system capabilities" could mention the version;

```console
DEBU[2024-12-13T15:19:59.621232216Z] capabilities requested                        spanID=6850a28621ab06d5 traceID=7367b3cf86d4c9e2d42e75f62eedc1ce version=123
DEBU[2024-12-13T15:19:59.621389882Z] fetching system capabilities                  spanID=6850a28621ab06d5 traceID=7367b3cf86d4c9e2d42e75f62eedc1ce
DEBU[2024-12-13T15:19:59.621546382Z] handling GET request                          method=GET module=api request-url="/_ping?capabilities=123" spanID=6850a28621ab06d5 status=200 traceID=7367b3cf86d4c9e2d42e75f62eedc1ce vars="map[]"
```